llvm/lib/Target/DirectX/DXILWriter/DXILBitcodeWriter.cpp

```diff
@@ -243,6 +243,10 @@ class DXILBitcodeWriter {
   }
   void writeDIDerivedType(const DIDerivedType *N,
                           SmallVectorImpl<uint64_t> &Record, unsigned Abbrev);
+  void writeDISubrangeType(const DISubrangeType *N,
```

Comment:
Is this a virtual function impl?  It'd be great if this file used "override". (Probably out of scope for this change)

Reply:
No its caused by `HANDLE_SPECIALIZED_MDNODE_LEAF_UNIQUABLE(DISubrangeType)` in `llvm/include/llvm/IR/Metadata.def` line 121. This is because `llvm/lib/Target/DirectX/DXILWriter/DXILBitcodeWriter.cpp` uses `Metadata.def` in `DXILBitcodeWriter::writeMetadataRecords`